File: examples/java-hello-world/src/test/java/org/organicprogramming/hello/HelloServiceTest.java
package org.organicprogramming.hello;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class HelloServiceTest {
    @Test
    void greetWithName() {
        assertEquals("Hello, Alice!", HelloService.greet("Alice"));
    }

    @Test
    void greetDefault() {
        assertEquals("Hello, World!", HelloService.greet(""));
    }

    @Test
    void greetNull() {
        assertEquals("Hello, World!", HelloService.greet(null));
    }
}

File: examples/hello-world/kotlin-hello-world/src/test/kotlin/org/organicprogramming/hello/HelloServiceTest.kt
package org.organicprogramming.hello

import kotlin.test.Test
import kotlin.test.assertEquals

class HelloServiceTest {
    @Test fun greetWithName() {
        assertEquals("Hello, Alice!", HelloService.greet("Alice"))
    }

    @Test fun greetDefault() {
        assertEquals("Hello, World!", HelloService.greet(""))
    }
}
